src/System.Text.Json/tests/Serialization/EnumConverterTests.cs

```diff
@@ -0,0 +1,117 @@
+        public void ConvertFileAttributes()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions();
+            options.Converters.Add(new JsonStringEnumConverter());
+
+            FileState state = JsonSerializer.Parse<FileState>(@"{""Attributes"":""ReadOnly""}", options);
+            Assert.Equal(FileAttributes.ReadOnly, state.Attributes);
+            state = JsonSerializer.Parse<FileState>(@"{""Attributes"":""Directory, ReparsePoint""}", options);
+            Assert.Equal(FileAttributes.Directory | FileAttributes.ReparsePoint, state.Attributes);
+            FileAttributes attributes = JsonSerializer.Parse<FileAttributes>(@"""Normal""", options);
+            Assert.Equal(FileAttributes.Normal, attributes);
+            attributes = JsonSerializer.Parse<FileAttributes>(@"""System, SparseFile""", options);
+            Assert.Equal(FileAttributes.System | FileAttributes.SparseFile, attributes);
+
+            // We are case insensitive on read
+            attributes = JsonSerializer.Parse<FileAttributes>(@"""OFFLINE""", options);
+            Assert.Equal(FileAttributes.Offline, attributes);
+            attributes = JsonSerializer.Parse<FileAttributes>(@"""compressed, notcontentindexed""", options);
+            Assert.Equal(FileAttributes.Compressed | FileAttributes.NotContentIndexed, attributes);
+
+            // Numbers are cool by default
+            attributes = JsonSerializer.Parse<FileAttributes>(@"131072", options);
+            Assert.Equal(FileAttributes.NoScrubData, attributes);
+            attributes = JsonSerializer.Parse<FileAttributes>(@"3", options);
+            Assert.Equal(FileAttributes.Hidden | FileAttributes.ReadOnly, attributes);
+
+            string json = JsonSerializer.ToString(FileAttributes.Hidden, options);
+            Assert.Equal(@"""Hidden""", json);
+            json = JsonSerializer.ToString(FileAttributes.Temporary | FileAttributes.Offline, options);
+            Assert.Equal(@"""Temporary, Offline""", json);
+
+            // Try a unique casing
+            options = new JsonSerializerOptions();
+            options.Converters.Add(new JsonStringEnumConverter(new ToLower()));
+
+            json = JsonSerializer.ToString(FileAttributes.NoScrubData, options);
+            Assert.Equal(@"""noscrubdata""", json);
+            json = JsonSerializer.ToString(FileAttributes.System | FileAttributes.Offline, options);
+            Assert.Equal(@"""system, offline""", json);
+
+            // Undefined values should come out as a number (not a string)
+            json = JsonSerializer.ToString((FileAttributes)(-1), options);
+            Assert.Equal(@"-1", json);
+
+            // Not permitting integers should throw
+            options = new JsonSerializerOptions();
+            options.Converters.Add(new JsonStringEnumConverter(allowIntegerValues: false));
+            Assert.Throws<FormatException>(() => JsonSerializer.ToString((FileAttributes)(-1), options));
```

Comment:
JsonException?

Reply:
Yes, I didn't catch up with the exception changes from yesterday- fixing.